sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -735,33 +768,94 @@ static void set_codec_init_func(const struct snd_soc_acpi_link_adr *link,
 	}
 }
 
+/*
+ * check endpoint status in slaves and gather link ID for all slaves in
+ * the same group to generate different CPU DAI. Now only support
+ * one sdw link with all slaves set with only single group id.
+ */
+static void get_slave_info(const struct snd_soc_acpi_link_adr *adr_link,
+			   int *dai_id, int *dai_num, int *codec_num,
+			   bool *group_generated)
+{
+	const struct snd_soc_acpi_adr_device *adr_d;
+	const struct snd_soc_acpi_link_adr *adr_next;
+	int group_id;
+	int index = 0;
+
+	*codec_num = adr_link->num_adr;
+	adr_d = adr_link->adr_d;
+
+	dai_id[index++] = ffs(adr_link->mask) - 1;
+	if (!adr_d->endpoints->aggregated) {
+		*dai_num = index;
```

Comment:
I would set *dai_num = 1 instead of index just to be explicit

Reply:
thanks, I will refine it.